### PR TITLE
fix: React Native CodeGen integration for 0.64-stable

### PR DESCRIPTION
**Pull request by grabbou**
## Summary

This PR resolves issues reported in the 0.64.0 release candidate around CodeGen. Further reading here: https://github.com/react-native-community/releases/issues/214#issuecomment-783363624

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -105,6 +105,7 @@
     "promise": "^8.0.3",
     "prop-types": "^15.7.2",
     "react-devtools-core": "^4.6.0",
+    "react-native-codegen": "^0.0.7",
```
Comment (grabbou) [inline package.json:108]: Note: On 0.64-stable branch, React Native will always use CodeGen from `npm`, even when running from RNTester. 

I wish we had Yarn Workspaces set up right, that way, I could point here to use local version of CodeGen when running from this repository.

Something we will have to improve in order to merge this to master.

CC: @hramos

File: scripts/generate-specs.sh [hunk 2]
```diff
@@ -53,33 +53,30 @@ main() {
...
   fi
 
-  if [ -d "$CODEGEN_REPO_PATH" ]; then
-    CODEGEN_PATH=$(cd "$CODEGEN_REPO_PATH" && pwd)
-  elif [ -d "$CODEGEN_NPM_PATH" ]; then
-    CODEGEN_PATH=$(cd "$CODEGEN_NPM_PATH" && pwd)
-  else
-    echo "Error: Could not determine react-native-codegen location. Try running 'yarn install' or 'npm install' in your project root." 1>&2
+  # Check if user-defined CODEGEN_PATH is correct
+  if [[ "$CODEGEN_PATH" && ! -d "$CODEGEN_PATH" ]]; then
```
Comment (grabbou) [inline scripts/generate-specs.sh:62]: One thing that I just realised is that you won't be able to set this unless the Script Phase is part of a user project. Right now, it's inside `FBReactNativeSpec` target.

So, I will most likely drop it.

File: package.json (hunk 1 above)
Comment (grabbou) [reply package.json:108]: In this particular case, this is more than fine. Right now, on 0.64-stable, in `packages`, we have CodeGen at 0.0.6 which doesn't work with this release. We would need to cherry-pick ALL commits related to CodeGen to bump it to 0.0.7.

By using an npm version instead, we can test RNTester and others. This isn't ideal setup, but works.

Comment (janicduplessis) [reply package.json:108]: Can you use `repo-config/package.json` to add the codegen dep? This one is a regular workspace.

**Pull request merged**